src/main/java/Duke.java

```diff
@@ -1,10 +1,158 @@
+        printTask(event);
+    }
+
+    public static void deadline(String cmd) {
+        int startPosition = cmd.indexOf(" ");
+        int timePosition = cmd.indexOf(" /by ");
+        String cmdDeadline = cmd.substring(startPosition + 1, timePosition);
+        String cmdTime = cmd.substring(timePosition + charLengthToSkip, cmd.length());
+        Task deadline = new Deadline(taskCounter, cmdDeadline, false, cmdTime);
+        tasks.add(deadline);
+        taskCounter++;
+        printTask(deadline);
+    }
+
+    public static void done(int taskNumber) {
+        tasks.get(taskNumber).setStatus();
+        System.out.println("    ╔═══════════════════════════════════════════════════════════╗");
+        print("Nice! I've marked this task as done: ");
+        print(tasks.get(taskNumber).getTaskType() + tasks.get(taskNumber).getStatusIcon() + " " + tasks.get(taskNumber).description);
+        System.out.println("    ╚═══════════════════════════════════════════════════════════╝");
+        System.out.println("    What can I do for you? Type 'bye' to exit.");
+        System.out.println();
+    }
+
+    public static void list() {
+        System.out.println("    ╔═══════════════════════════════════════════════════════════╗");
+        print("Here are the tasks in your list:");
+        for(Task t: tasks){
+            if(t == null){
+                break;
+            }
+            print(t.toString());
+        }
+        System.out.println("    ╚═══════════════════════════════════════════════════════════╝");
+        System.out.println("    What can I do for you? Type 'bye' to exit.");
+        System.out.println();
+    }
+
+    public static void exit() {
+        System.out.println("    ╔═══════════════════════════════════════════════════════════╗");
+        print("Bye! See you next time :)");
+        System.out.println("    ╚═══════════════════════════════════════════════════════════╝");
+        System.exit(0);
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        greet();
+
+        // read command-line input
+        Scanner scanner = new Scanner(System.in);
+        String cmd = scanner.nextLine();
+
+        while(!cmd.equals("bye")) {
+            if (cmd.equals("list")) {
+                list();
+            } else if (cmd.length() > 4 && cmd.substring(0, 4).equals("todo")) {
+                todo(cmd);
+            } else if (cmd.length() > 5 && cmd.substring(0, 5).equals("event")) {
+                event(cmd);
+            } else if (cmd.length() > 8 && cmd.substring(0, 8).equals("deadline")) {
+                deadline(cmd);
+            } else if (cmd.length() > 4 && cmd.substring(0, 4).equals("done")) {
```

Comment:
Maybe consider checking for the type of command first before checking input length, such that if the command is invalid, we will know why it is invalid (such as if the command is correct but empty or the command is invalid/misspelled). This is especially for Level-5 when we need to differentiate between exceptions.

Reply:
I use switch to handle different commands which make this part more clear, but your advice is also helpful.